src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5188,6 +5193,250 @@ private static List<CompletionResult> CompleteRequires(CompletionContext context
+            {
+                return null;
+            }
+
+            // Last keyword at or before the cursor
+            Match lineKeyword = null;
+            foreach (Match keyword in usedKeywords)
+            {
+                if (context.CursorPosition.Offset >= keyword.Index + context.TokenAtCursor.Extent.StartOffset)
+                {
+                    lineKeyword = keyword;
+                }
+            }
+            if (lineKeyword is null)
+            {
+                return null;
+            }
+
+            // Cursor is within or at the start/end of the keyword
+            if (context.CursorPosition.Offset <= lineKeyword.Index + lineKeyword.Length + context.TokenAtCursor.Extent.StartOffset)
+            {
+                replacementIndex = context.TokenAtCursor.Extent.StartOffset + lineKeyword.Index;
+                replacementLength = lineKeyword.Value.Length;
+
+                var validKeywords = new HashSet<String>(s_commentHelpKeywords.Keys, StringComparer.OrdinalIgnoreCase);
+                foreach (Match keyword in usedKeywords)
+                {
+                    if (s_commentHelpAllowedDuplicateKeywords.Contains(keyword.Value) || keyword == lineKeyword)
+                    {
+                        continue;
+                    }
+
+                    validKeywords.Remove(keyword.Value);
+                }
+
+                var result = new List<CompletionResult>();
+                foreach (string keyword in validKeywords)
+                {
+                    if (keyword.StartsWith(lineKeyword.Value, StringComparison.OrdinalIgnoreCase))
+                    {
+                        result.Add(new CompletionResult(keyword, keyword, CompletionResultType.Keyword, s_commentHelpKeywords[keyword]));
+                    }
+                }
+
+                return result;
+            }
+
+            // Finds the argument for the keyword (any characters following the keyword, ignoring leading/trailing whitespace). For example "C:\New folder"
+            Match keywordArgument = Regex.Match(context.CursorPosition.Line, @"(?<=^\s*\.\w+\s+)\S.*(?<=\S)");
+            int lineStartIndex = lineKeyword.Index - context.CursorPosition.Line.IndexOf(lineKeyword.Value) + context.TokenAtCursor.Extent.StartOffset;
+            int argumentIndex = keywordArgument.Success ? keywordArgument.Index : context.CursorPosition.ColumnNumber - 1;
+
+            replacementIndex = lineStartIndex + argumentIndex;
+            replacementLength = keywordArgument.Value.Length;
+
+            if (lineKeyword.Value.Equals("PARAMETER", StringComparison.OrdinalIgnoreCase))
+            {
+                return CompleteCommentParameterValue(context, keywordArgument.Value);
+            }
+
+            if (lineKeyword.Value.Equals("FORWARDHELPTARGETNAME", StringComparison.OrdinalIgnoreCase))
+            {
+                return new List<CompletionResult>(CompleteCommand(keywordArgument.Value, "*", CommandTypes.All));
```

Comment:
```suggestion
                return CompleteCommand(keywordArgument.Value, "*", CommandTypes.All);
```

Reply:
This won't compile for me because of:

"CS0266	Cannot implicitly convert type 'System.Collections.Generic.IEnumerable<System.Management.Automation.CompletionResult>' to 'System.Collections.Generic.List<System.Management.Automation.CompletionResult>'. An explicit conversion exists (are you missing a cast?)

Reply:
Ah, I see. Adding the new list constructor is fine for now. We should go back through this code and unify the return types and see if we can return null in places where we would otherwise have an empty list.

Reply:
I've started by fixing my code at least so it will always return a populated list or null.